bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java

```diff
@@ -86,15 +83,10 @@ public Object invokeMethodAsync(BObject object, String methodName, String strand
      */
     public void invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
                                   Callback callback, Map<String, Object> properties, Object... args) {
-        Function<Object[], Object> func = objects -> {
-            Strand strand = (Strand) objects[0];
-            if (ObserveUtils.isObservabilityEnabled() && properties != null &&
-                    properties.containsKey(ObservabilityConstants.KEY_OBSERVER_CONTEXT)) {
-                strand.observerContext =
-                        (ObserverContext) properties.remove(ObservabilityConstants.KEY_OBSERVER_CONTEXT);
-            }
-            return object.call(strand, methodName, args);
-        };
+        if (object == null) {
+            throw new NullPointerException();
```

Comment:
This is not necessary. Is it? Because in the next line it will automatically throw Nullpointer and throwing this exception explicitly is not recommended AFAIK

Reply:
below `object.call` is in a lambda, therefor error gets thrown in a different thread. so the original stack trace is lost. 
Explicit null pointer exception is ok for parameter checking. eg: http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Objects.java#l201

Reply:
Reading on this, came across this: https://stackoverflow.com/questions/3322638/is-it-okay-to-throw-nullpointerexception-programmatically 
It seems to be a point of argument 😀